File: src/01-easy/pirates-treasure/solution.pl
$| = 1;
my $w = <STDIN>+0; my $h = <STDIN>+0;
my @g; for (0..$h-1) { push @g, [split / /, <STDIN>]; }
my @dx=(-1,-1,-1,0,0,1,1,1); my @dy=(-1,0,1,-1,1,-1,0,1);
for my $y (0..$h-1) { for my $x (0..$w-1) { if ($g[$y][$x]==0) {
    my $ok=1; for my $i (0..7) { my($nx,$ny)=($x+$dx[$i],$y+$dy[$i]);
        $ok=0 if $nx>=0&&$nx<$w&&$ny>=0&&$ny<$h&&$g[$ny][$nx]!=1; }
    if ($ok) { print "$x $y\n"; exit; }
}}}

File: src/01-easy/pirates-treasure/solution.dart
import 'dart:io';
void main(){
    int w=int.parse(stdin.readLineSync()!),h=int.parse(stdin.readLineSync()!);
    var g=List.generate(h,(_)=>stdin.readLineSync()!.trim().split(' ').map(int.parse).toList());
    var dx=[-1,-1,-1,0,0,1,1,1],dy=[-1,0,1,-1,1,-1,0,1];
    for(int y=0;y<h;y++) for(int x=0;x<w;x++) if(g[y][x]==0){
        bool ok=true;for(int i=0;i<8;i++){int nx=x+dx[i],ny=y+dy[i];if(nx>=0&&nx<w&&ny>=0&&ny<h&&g[ny][nx]!=1)ok=false;}
        if(ok){print('$x $y');return;}
    }
}
